nucypher/network/server.py

```diff
@@ -267,37 +273,41 @@ def reencrypt_via_rest(self, hrac_as_hex, request: http.Request):
 
         return Response(content=cfrag_byte_stream, headers=headers)
 
-    def provide_treasure_map(self, treasure_map_id_as_hex):
-        # For now, grab the TreasureMap for the DHT storage.  Soon, no do that.  #TODO!
-        treasure_map_id = binascii.unhexlify(treasure_map_id_as_hex)
+    def provide_treasure_map(self, treasure_map_id):
         headers = {'Content-Type': 'application/octet-stream'}
 
         try:
-            treasure_map_bytes = self.dht_server.storage[digest(treasure_map_id)]
-            response = Response(content=treasure_map_bytes, headers=headers)
+            treasure_map = self.treasure_maps[digest(treasure_map_id)]
+            response = Response(content=bytes(treasure_map), headers=headers)
+            self.log.info("{} providing TreasureMap {}".format(self, treasure_map_id))
         except KeyError:
+            self.log.info("{} doesn't have requested TreasureMap {}".format(self, treasure_map_id))
             response = Response("No Treasure Map with ID {}".format(treasure_map_id),
                                 status_code=404, headers=headers)
 
         return response
 
-    def receive_treasure_map(self, treasure_map_id_as_hex, request: http.Request):
-        # TODO: This function is the epitome of #172.
-        treasure_map_id = binascii.unhexlify(treasure_map_id_as_hex)
-
-        header, signature_for_ursula, pubkey_sig_alice, ether_address, hrac, tmap_message_kit = \
-            dht_with_hrac_splitter(request.body, return_remainder=True)
-        # TODO: This next line is possibly the worst in the entire codebase at the moment.  #172.
-        # Also TODO: TTL?
-        do_store = self.dht_server.protocol.determine_legality_of_dht_key(
-                    signature_for_ursula, pubkey_sig_alice,
-                    hrac, digest(treasure_map_id), request.body)
+    def receive_treasure_map(self, treasure_map_id, request: http.Request):
+        from nucypher.policy.models import TreasureMap
+
+        try:
+            treasure_map = TreasureMap.from_bytes(
+                bytes_representation=request.body,
+                verify=True)
```

Comment:
A question/comment for later -- does `verify=True` cause this to error if it cannot be verified?

Reply:
Yep, as you can see it hits this line:

`raise self.InvalidPublicSignature("This TreasureMap is not properly publicly signed by Alice.")`

...I've started transitioning all verification logic to this pattern.